### PR TITLE
Support round-trip CRD (de)serialization

### PR DESCRIPTION
**Pull request by stan-sz**
Support round-trip CRD (de)serialization

### CODE REVIEW

**Comment by stan-sz** [comment]
Fixes #1032

File: tests/KubernetesClient.Tests/KubernetesYamlTests.cs [hunk 1]
```diff
@@ -841,5 +841,56 @@ public void DeserializeWithJsonPropertyName()
...
+#pragma warning disable CA1812 // Class is used for YAML deserialization tests
+        [KubernetesEntity(Group = KubeGroup, Kind = KubeKind, ApiVersion = KubeApiVersion, PluralName = KubePluralName)]
+        private sealed class V1AlphaFoo : IKubernetesObject<V1ObjectMeta>, ISpec<Dictionary<string, object>>
+        {
+            public const string KubeApiVersion = "v1alpha";
+            public const string KubeKind = "foo";
+            public const string KubeGroup = "foo.bar";
+            public const string KubePluralName = "foos";
+
+            public string? ApiVersion { get; set; }
```
Comment (tg123) [inline tests/KubernetesClient.Tests/KubernetesYamlTests.cs:854]: could you please fix those warnings

**Comment by tg123** [comment]
nit warnings, then LGTM

File: tests/KubernetesClient.Tests/KubernetesYamlTests.cs (hunk 1 above)
Comment (stan-sz) [reply tests/KubernetesClient.Tests/KubernetesYamlTests.cs]: Done

**Review by tg123 — APPROVED** [review]
/LGTM

**Pull request merged**